lib/pbench/server/api/resources/auth.py

```diff
@@ -0,0 +1,88 @@
+import jwt
+import os
+import datetime
+from flask import request, abort
+from flask_httpauth import HTTPTokenAuth
+from pbench.server.api.resources.models import UserModel, ActiveTokens
+from pbench.server.api.resources.database import Database
+
+
+auth = HTTPTokenAuth("Bearer")
+
+
+def encode_auth_token(token_expire_duration, user_id):
+    """
+    Generates the Auth Token
+    :return: string
+    """
+    payload = {
+        "iat": datetime.datetime.utcnow(),
+        "exp": datetime.datetime.utcnow()
+        + datetime.timedelta(minutes=int(token_expire_duration)),
+        "sub": user_id,
+    }
+
+    # Get jwt key
+    jwt_key = get_secret_key()
+    return jwt.encode(payload, jwt_key, algorithm="HS256")
+
+
+@auth.verify_token
+def verify_token(auth_token):
+    """
+    Validates the auth token
+    :param auth_token:
+    :param app:
+    :return: integer|string
+    """
+    try:
+        payload = jwt.decode(auth_token, get_secret_key(), algorithms="HS256")
+        user_id = payload["sub"]
+        if ActiveTokens.valid(auth_token):
+            user = UserModel.query.filter_by(id=user_id).first()
+            return user
+        return False
+    except jwt.ExpiredSignatureError:
+        ActiveTokens.query.filter_by(token=auth_token).delete()
+        Database.db_session.commit()
+        return False
+    except jwt.InvalidTokenError:
+        return False
+    except Exception:
+        return False
+
+
+def get_secret_key():
+    try:
+        return os.getenv("SECRET_KEY", "my_precious")
+    except Exception as e:
+        print(f"{__name__}: ERROR: {e.__traceback__}")
+
+
+def get_auth_token(logger):
+    # get auth token
+    auth_header = request.headers.get("Authorization")
+
+    if not auth_header:
+        logger.error("Malformed Authorization header")
```

Comment:
This is not a server side error, so we should either emit it as a `warning` or as an `info` message.